clean_project/search_api.py

```diff
@@ -0,0 +1,45 @@
+"""
+Search a word in a file and return all the occurrences specified the line numbers
+"""
+import argparse
+from collections import defaultdict
+from flask import Flask, jsonify
+import search_engine
+
+BOLD = '\033[1m'
+END = '\033[0m'
+
+
+app = Flask(__name__)
+
+
+def config_app(search_object):
+    app.config['search'] = search_object
+    return app
+
+
+@app.route('/search/<string:search_word>', methods=['GET'])
+def get_request(search_word):
+    """
+    Get request to verify if a word exist in a database
+    :param search_word: Word used for search
+    :return: Response of the search request
+    """
+    save_result = defaultdict(lambda: defaultdict(list))
+    result_search = app.config['search'].search(search_word)
+    if result_search is not []:
```

Comment:
i told you this already. it is not idiomatic to check if the search result `is not []`. you should check the length, or simply `if search_result`, which evaluates to a boolean anyway